packages/features/users/components/UserTable/UserListTable.tsx

```diff
@@ -0,0 +1,353 @@
+            canRemove: permissionsRaw.canRemove && !isSelf,
+            canImpersonate: user.accepted && !user.disableImpersonation && !isSelf,
+            canLeave: user.accepted && isSelf,
+          };
+
+          return <TableActions user={user} permissionsForUser={permissionsForUser} dispatch={dispatch} />;
+        },
+      },
+    ];
+
+    return cols;
+  }, [session?.user.id, adminOrOwner, dispatch]);
+
+  //we must flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(() => data?.pages?.flatMap((page) => page.rows) ?? [], [data]) as User[];
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  const totalFetched = flatData.length;
+
+  //called on scroll and possibly on mount to fetch more data as the user scrolls and reaches bottom of table
+  const fetchMoreOnBottomReached = useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
+        if (scrollHeight - scrollTop - clientHeight < 300 && !isFetching && totalFetched < totalDBRowCount) {
+          fetchNextPage();
+        }
+      }
+    },
+    [fetchNextPage, isFetching, totalFetched, totalDBRowCount]
+  );
+
+  useEffect(() => {
+    fetchMoreOnBottomReached(tableContainerRef.current);
+  }, [fetchMoreOnBottomReached]);
+
+  const removeMemberMutation = trpc.viewer.teams.removeMember.useMutation({
+    async onSuccess() {
+      await utils.viewer.teams.get.invalidate();
+      await utils.viewer.eventTypes.invalidate();
+      await utils.viewer.organizations.listMembers.invalidate();
+      showToast(t("success"), "success");
+    },
+    async onError(err) {
+      showToast(err.message, "error");
+    },
+  });
+
+  return (
+    <>
+      <DataTable
+        searchKey="member"
+        selectionOptions={[
+          {
+            label: "Add To Team",
+            onClick: () => {
+              console.log("Add To Team");
+            },
+            icon: Users,
+          },
+          {
+            label: "Delete",
+            onClick: () => {
+              console.log("Delete");
+            },
+            icon: StopCircle,
+          },
+        ]}
+        tableContainerRef={tableContainerRef}
+        tableCTA={
+          adminOrOwner && (
+            <Button
+              type="button"
+              color="primary"
+              StartIcon={Plus}
+              size="sm"
+              className="rounded-md"
+              onClick={() =>
+                dispatch({
+                  type: "INVITE_MEMBER",
+                  payload: {
+                    showModal: true,
+                  },
+                })
+              }
+              data-testid="new-organization-member-button">
+              {t("add")}
+            </Button>
```

Comment:
We inline the CTA for this page (used to be in META) but we make it fit with the filters used in app